source/MaterialXRender/Handlers/StbImageLoader.cpp

```diff
@@ -145,6 +145,8 @@ bool StbImageLoader::acquireImage(const FilePath& filePath, ImageDesc &imageDesc
         imageDesc.height = iheight;
         imageDesc.channelCount = ichannelCount;
         imageDesc.computeMipCount();
+        // Set the deallocator to be the one provided with the library
+        imageDesc.resourceBufferDeallocator = &stbi_image_free;
```

Comment:
STB image loader deallocator. Can just point to function in library since signature matches.